### DISCUSSION TITLE
android termux

### DISCUSSION THREAD

**Discussion by sl33pyC01E** [leejet/stable-diffusion.cpp#205] [Ideas]
Hello, is there an easy way to port this to termux on android?
(I haven't begun attempting yet)

**Comment by dwalter** [comment]
Yes, I got this to work on termux on android on a Samsung galaxy s23 by just following the README without doing anything special and it just worked.


At default settings I'm getting ~70s/it for 20 iterations of stable diffusion so this'll take about 20-25 minutes to generate a single image. Assuming this is not leveraging the GPU with my current settings.

**Reply by sl33pyC01E** [reply]
i have gotten it to complete an image in 97 seconds with 512x512 sdt lcm

**Comment by patientx** [comment]
GUIDE FROM ZERO ON AN ANDROID PHONE - TERMUX AND WHOLE SETUP
---
For someone coming from zero on an android phone 👍

1- ) Install termux from f-droid. (https://f-droid.org/packages/com.termux/)
---- Open termux & enter and execute these commands one by one
2- ) pkg update
3- ) termux-setup-storage
4- ) pkg install git cmake -y
5- ) git clone --recursive https://github.com/leejet/stable-diffusion.cpp
6-) cd stable-diffusion.cpp
7-) curl -L -O https://huggingface.co/CompVis/stable-diffusion-v-1-4-original/resolve/main/sd-v1-4.ckpt
 or just download any fp16 model which are a lot of huggingface and only 2gb instead of default 4gb
8-) mkdir build
9-) cd build
10-) cmake ..
11-) cmake --build . --config Release

and run.

**Reply by SA-j00u** [reply]
can termux packages by installed in offline mode?
it have access to whole (no root) filesystem with no root?

**Reply by patientx** [reply]
From the termux webpage ::

You can access Termux home directory ($HOME) from the file manager using Storage Access Framework and capable of accessing drives like USB or external SD-card in read-write mode.

Few recommended file managers which can access Termux home directory:

FX File Explorer: https://play.google.com/store/apps/details?id=nextapp.fx
Material Files: https://play.google.com/store/apps/details?id=me.zhanghai.android.files

![300px-FX_Termux_Home](https://github.com/leejet/stable-diffusion.cpp/assets/5392772/95b741ff-c770-4b7e-8a17-8762ce650c85)


I am using material files and yes it is very easy to copy paste move ... files from there to normal phone memory or just view output etc.

**Reply by SA-j00u** [reply]
hmm wine root directory can't be accessed in this way?

**Comment by patientx** [comment]
> can termux packages by installed in offline mode? it have access to whole (no root) filesystem with no root?

it "emulates" root, so no it won't have access to whole system but the apps you install over it are made to be believe they are doing that. for example you can install proot in it and install ubuntu over it too. And I don't think you can do installs in offline mode.

**Reply by SA-j00u** [reply]
> it won't have access to whole system

i'm about user files
can it run own apps, for example, from sdcard root folder?
and
is it's "root files" stored in accessible folder?
for copying preinstalled packages

also why not make prebuild SD for only copying it to termux?

**Reply by SA-j00u** [reply]
i post my prebuilt binary
https://github.com/leejet/stable-diffusion.cpp/issues/231
for less advanced users
